relation_engine_server/utils/wait_for.py

```diff
@@ -5,66 +5,39 @@
 import time
 import sys
 from relation_engine_server.utils.config import get_config
+from typing import List
 
 _CONF = get_config()
 
 
-def wait_for_service(service_list):
+def wait_for_service(service_list: List[str]) -> None:
     '''wait for a service or list of services to start up'''
     timeout = int(time.time()) + 60
-
-    service_conf_list = [get_service_conf(s) for s in service_list]
-
-    while True:
-        try:
-            for service in service_conf_list:
-                name = service['name']
-                url = service['url']
-                if service['auth'] is not None:
-                    requests.get(service['url'], auth=service['auth']).raise_for_status()
-                else:
-                    # auth and workspace both return 500, so don't raise_for_status
-                    requests.get(service['url'])
-            break
-        except Exception:
-            print(f"Waiting for {name} to start...")
-            if int(time.time()) > timeout:
-                raise RuntimeError(f"Timed out waiting for {name}, {url}")
-            time.sleep(3)
-    print(f"{name} started!")
-
-
-def get_service_conf(service_name):
-
-    service_conf = {
-        'arangodb': {
-            'url': _CONF['db_url'],
-        },
-        'auth': {
-            'url': _CONF['auth_url'],
-        },
-        'workspace': {
-            'url': _CONF['workspace_url'],
-        },
-        'localhost': {
-            'url': 'http://127.0.0.1:5000',
-        }
-    }
-
-    if service_name not in service_conf:
-        raise KeyError(f'Configuration for {service_name} not found')
-
-    return {
-        'name': service_name,
-        # auth defaults to None if there is nothing set
-        'auth': service_conf[service_name].get('auth'),
-        'url': service_conf[service_name]['url'],
-    }
+    services_pending = set(service_list)
```

Comment:
To get it to work more reliably waiting for arangodb, I added the 'callback' option for the services, and in the process refactored things a little bit and added the 'raise_for_status' option.